File: apps/mobile/plugins/LunaraWidgetModule.swift
import Foundation
import WidgetKit

@objc(LunaraWidgetModule)
class LunaraWidgetModule: NSObject {

  @objc func updateWidget(_ data: NSDictionary) {
    let defaults = UserDefaults(suiteName: "group.com.shinracode.lunara")
    defaults?.set(data["phase"] as? String ?? "Fase lútea", forKey: "phase")
    defaults?.set(data["phaseEmoji"] as? String ?? "🌙",    forKey: "phaseEmoji")
    defaults?.set(data["dayOfCycle"] as? Int ?? 1,          forKey: "dayOfCycle")
    defaults?.set(data["daysUntilPeriod"] as? Int ?? -1,    forKey: "daysUntilPeriod")
    defaults?.synchronize()

    if #available(iOS 14.0, *) {
      WidgetCenter.shared.reloadAllTimelines()
    }
  }

  @objc static func requiresMainQueueSetup() -> Bool { false }
}

File: apps/mobile/plugins/LunaraWidgetModule.kt
package com.shinracode.lunara

import android.appwidget.AppWidgetManager
import android.content.ComponentName
import android.content.Context
import android.content.SharedPreferences
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.ReadableMap
import com.shinracode.lunara.widget.LunaraWidgetProvider

class LunaraWidgetModule(private val reactContext: ReactApplicationContext) :
    ReactContextBaseJavaModule(reactContext) {

    override fun getName(): String = "LunaraWidgetModule"

    @ReactMethod
    fun updateWidget(data: ReadableMap) {
        val prefs: SharedPreferences = reactContext
            .getSharedPreferences("LunaraWidget", Context.MODE_PRIVATE)
            .edit()
            .apply {
                putString("phase", data.getString("phase"))
                putString("phaseEmoji", data.getString("phaseEmoji"))
                putInt("dayOfCycle", data.getInt("dayOfCycle"))
                putInt("daysUntilPeriod", data.getInt("daysUntilPeriod"))
            }
            .let { it.apply(); reactContext.getSharedPreferences("LunaraWidget", Context.MODE_PRIVATE) }

        val manager = AppWidgetManager.getInstance(reactContext)
        val ids = manager.getAppWidgetIds(
            ComponentName(reactContext, LunaraWidgetProvider::class.java)
        )
        ids.forEach { LunaraWidgetProvider.updateWidget(reactContext, manager, it) }
    }
}
